libs/muddle/src/muddle.cpp

```diff
@@ -93,7 +93,7 @@ Muddle::Muddle(NetworkId network_id, CertificatePtr certificate, NetworkManager
 
 Muddle::~Muddle()
 {
-  MuddleRegistry::Instance().Unregister(node_address_);
+  MuddleRegistry::Instance().Unregister(this);
```

Comment:
I am curious about what this does

Reply:
Basically just removes an element from an internal map. The internal map contains weak pointers to muddle instances. The key for the map is the pointer to the instance (and that is all that it is used for).